help/gory_details/_posts/2021-04-26-author-search.md

```diff
@@ -0,0 +1,103 @@
+
+The library will recognize `surname` and `first name` most of the times correctly. So `Pope John Paul the II` would loose all his tiara and become `Paul, John` -- but we don't know anything besides the characters. We are not consulting wikipedia to find out the real name of the Pope, or the author (at least not yet). And our `Peter` is recognized as the first name, and `Adamczuk` as given.
+
+Next are some boring filters: one of them normalizes the names. `Adamczuk,               P` becomes `Adamczuk, P`, and `Lao'tzu` becomes `Lao tzu,`. And we also reject any search with 6 or more names. Why? Well, because it can lead to 6^n combinations (in the worst case) in the later stages.
+
+Next comes **transliterations** -- what's that you ask? Well, English is the Lingua Franca of scientific discourse - but only for the past 2 hundred years. The dominance of US alphabet in all things electronic is also thing of the past, but humans are forever lazy (or intelligent) and when confronted with `e`, `é`, `e` .... characters, they say 'Aha! That's `e`'. All accents lost. So naturally, they want to be able to search for `jonas` and find `jonáš` as well. And so we oblige. Internally at ADS we use `UTF-8` character encoding but we don't force the English-centric writers to think in more than 26 characters of English alphabet. Also by convention, scientific publications oftentimes insist on transliterating native author names into ASCII.
+
+So in the case of `adamczuk, p` at this stage we will discover that ADS database contains `adamčuk, peter` (we'd see also `adamčuk, p` if it was necessary)
+
+
+### Intermezzo #0 - how are transliterations generated/found?
+
+Periodically, small minions at ADS will wake up every week, read all author names and compile a list of the following form:
+
+```
+adamczuk, p => adamčuk, p
+adamczuk, peter => adamčuk, peter
+adamczuk, pavel => adamčuk, pavel
+```
+
+That is, we have a set of rules - some of them custom to ADS, and majority stemming from Java Unidecode library (convention) which will transform any non-ascii input into ascii **transliterations**. The generated vocabulary of names has at this point millions of entries (2024917 and counting) and contains every non-latin character that ever appeared in the ADS database. The dictionary is then used (at search time) to lookup `UTF-8` variants when users give us `ASCII` name.
+
+Technical note: the lookup is done via suffix trie (the datastructure is built by a finite state machine). So even though the dictionary has over 2M distinct entries, the cost of lookup is just the length of the input and the space occupied by the internal datastructure which is a fraction of the original size - if we were counting characters. Our 2M dictionary of names seems large, but still far cry from the theoretical limit of 2B nodes. For details see [http://blog.mikemccandless.com/2010/12/using-finite-state-transducers-in.html]. 
```

Comment:
* the parenthetical explaining trie is still confusing for a lay-person. Maybe "datastructure" -> "search tree" and remove "is" 
* "still far cry" -> "still a far cry"
* fix the link